apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -280,6 +319,22 @@ def __repr__(self):
             self._data,
         )
 
+    def find_type(self, keys):
+        if isinstance(keys, str):
+            key_list = keys.split(".")
+        elif not isinstance(keys, list):
+            key_list = list(keys)
+
+        schema_keys = list(chain.from_iterable(zip(['properties'] * len(key_list), key_list)))
```

Comment:
Possible alternative you can consider `list(chain.from_iterable(zip_longest((), key_list, fillvalue='properties')))`